### ISSUE TITLE
Gmail SMTP Sign-in Attempt Blocked

### ISSUE THREAD

**Issue by PatLuzny** [WPManageNinja/fluent-smtp#6]
I am attempting to use Fluent SMTP with Gmail's SMTP server.  While I've managed to set up the connection, the email fails to send as Google blocks the login attempt from Fluent SMTP using the credentials I provided in the connection.

There is an option in Google's Admin Console to "allow less secure apps" which should make this work, but Google advises against it, and seems to be moving towards deprecating this [feature](https://support.google.com/a/answer/6260879?hl=en#zippy=%2Cnotice-of-setting-removal%2Cif-you-allow-sign-ins-from-less-secure-apps): "Because Google is beginning to shut off Google Account access to less secure apps, the enforcement option is no longer available. We recommend turning off less secure apps access now. You should start using alternatives to less secure apps."

Is there a workaround to this for using Fluent SMTP with smtp.gmail.com, or a configuration option I am missing?

![Capture](https://user-images.githubusercontent.com/18220249/105795291-9798c900-5f5a-11eb-9a91-7a194e5a4efe.PNG)

**Comment by PatLuzny** [comment]
To elaborate on this, I see that the issue is that we need to use OAuth 2.0 to authenticate instead of just a username and password. This is possible in Postman SMTP and is outlined in this [article](https://www.cloudways.com/blog/post-smtp-mailer-fork-of-wordpress-postman-smtp-plugin/?id=339490#step10).  If I enter my client ID and secret in the SMTP username and password fields in Fluent SMTP, testing the connection gives me the following error: "SMTP Error: Could not authenticate."

Are there plans to add OAuth 2.0 authentication?

**Comment by TimGadget** [comment]
When I added Zoho Mail I used a password generated for connection to apps. I know that gmail has this feature too. Have you tried this? Certainly more secure than using the password for your email (especially with the issue that I opened too).

**Comment by zocoloco** [comment]
Same here. I need OAuth 2.0. Are you planning to add this? In the meantime, I'll continue to use Post SMTP

**Comment by Alexander-van-aken** [comment]
A connection provider for Google Workspace would be awesome with a note link for the Authorized redirect URIs and input fields for the Client ID and Client secret.

**Comment by killerb77** [comment]
Yes, please add this! Would be great to be able to use google workspace with this plugin.

**Comment by pramodraam** [comment]
This can be fixed by turning on low secure apps.
